subsys/bluetooth/mesh/keys.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_SUBSYS_BLUETOOTH_MESH_KEYS_H_
+#define ZEPHYR_SUBSYS_BLUETOOTH_MESH_KEYS_H_
+
+#include <string.h>
+#include <stdint.h>
+
+typedef uint8_t raw_key_t[16];
+
+enum key_type {
+	ECB_KEY,
+	CCM_KEY
+};
+
+#if defined CONFIG_BT_MESH_USES_MBEDTLS_PSA
+#include <psa/crypto.h>
+
+#define mesh_key_t psa_key_id_t
+#define GET_KEY_REF(key) &key
+
+int bt_mesh_import_key(enum key_type type, const raw_key_t in, mesh_key_t *out);
```

Comment:
I think the implementation should belong to `crypto_tc.c` and `crypto_mtls.c`.

Reply:
Moved implementation related to mbedtls into `crypto_mtls.c`. Left inline fucntions related to tinycrypt in header. There is no sense to move it in the c file since it will create jus recall `memcpy` or `memcmp`.